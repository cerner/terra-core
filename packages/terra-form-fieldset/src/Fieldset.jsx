/* eslint-disable react/jsx-boolean-value, jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import styles from './Fieldset.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Children the Field contains.
   */
  children: PropTypes.node,
  /**
   * Help element to display with other elements.
   */
  help: PropTypes.node,
  /**
   * Legend for the input group.
   */
  legend: PropTypes.string,
  /**
   * Attributes to attach to the legend.
   */
  // eslint-disable-next-line react/forbid-prop-types
  legendAttrs: PropTypes.object,
  /**
   * Determines whether the fieldset is required.
   */
  required: PropTypes.bool,
  /**
   * Whether or not the legend is visible. Use this prop to hide a legend while still creating it on the DOM for accessibility.
   */
  isLegendHidden: PropTypes.bool,
};

const defaultProps = {
  legendAttrs: {},
  required: false,
  isLegendHidden: false,
};

const Fieldset = ({
  children, help, legend, legendAttrs, isLegendHidden, required, ...customProps
}) => {
  const fieldsetClasses = cx([
    'fieldset',
    { 'fieldset-required': required },
    customProps.className,
  ]);

  const legendClasses = cx([
    'legend',
    legendAttrs.className,
    { 'legend-visually-hidden': isLegendHidden },
  ]);

  return (
    <fieldset {...customProps} className={fieldsetClasses}>
      {legend && <legend {...legendAttrs} className={legendClasses}>{legend}</legend>}
      {help && <small className={cx('help-text')} tabIndex="-1">{help}</small>}
      <div className={cx('fieldset-children')}>
        {children}
      </div>
    </fieldset>
  );
};

Fieldset.propTypes = propTypes;
Fieldset.defaultProps = defaultProps;

export default Fieldset;
