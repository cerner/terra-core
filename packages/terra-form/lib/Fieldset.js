'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

require('./Field.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /* eslint-disable react/jsx-boolean-value, jsx-a11y/label-has-for */


var propTypes = {
  /**
   * Children the Field contains
   */
  children: _propTypes2.default.node,
  /**
   * Error message for when the input is invalid
   */
  error: _propTypes2.default.node,
  /**
   * Help element to display with the input
   */
  help: _propTypes2.default.node,
  /**
   * Determines whether the fieldset is an inline fieldset
   */
  isInline: _propTypes2.default.bool,
  /**
   * Legend for the input group
   */
  legend: _propTypes2.default.string,
  /**
   * Determines whether the fieldset is required
   */
  required: _propTypes2.default.bool
};

var defaultProps = {
  children: null,
  error: null,
  help: null,
  isInline: false,
  legend: null,
  required: false
};

var Fieldset = function Fieldset(_ref) {
  var children = _ref.children,
      error = _ref.error,
      help = _ref.help,
      isInline = _ref.isInline,
      legend = _ref.legend,
      required = _ref.required,
      customProps = _objectWithoutProperties(_ref, ['children', 'error', 'help', 'isInline', 'legend', 'required']);

  var fieldsetClasses = (0, _classnames2.default)('terra-Form-fieldset', { 'terra-Form-fieldset--inline': isInline }, { 'terra-Form-fieldset--required': required }, customProps.className);

  return _react2.default.createElement(
    'fieldset',
    _extends({}, customProps, { className: fieldsetClasses }),
    legend && _react2.default.createElement(
      'legend',
      { className: 'terra-Form-legend' },
      legend
    ),
    children,
    help && _react2.default.createElement(
      'small',
      { className: 'terra-Form-helpText', tabIndex: '-1' },
      help
    ),
    error && _react2.default.createElement(
      'small',
      { className: 'terra-Form-error', tabIndex: '-1' },
      error
    )
  );
};

Fieldset.propTypes = propTypes;
Fieldset.defaultProps = defaultProps;

exports.default = Fieldset;