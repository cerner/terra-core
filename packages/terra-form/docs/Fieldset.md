# Fieldset

Generic form fieldset component which handles the layout of a standard form fieldset including errors, help text, legend, value and widget placement.

## Usage

```jsx
import React from 'react';
import Fieldset from 'terra-form/lib/Fieldset';

<Fieldset
  legend="Do you have any Children?"
  error="This field is required"
  help="Families are eligible for family package plans"
  required
  isInline
 >
   {this.props.children}
</Fieldset>
```
