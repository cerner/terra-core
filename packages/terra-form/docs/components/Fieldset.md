# Fieldset

Generic form fieldset component which handles the layout of a standard form fieldset including errors, help text, label, value and widget placement.

## Usage

```jsx
import React from 'react';
import Fieldset from 'terra-form/lib/Fieldset';

<Fieldset
  type="checkbox"
  legend="Do you have any Children?"
  name="children_present"
  value="children_present"
  error="This field is required"
  help="Families are eligible for family package plans"
  required
  isInline
 >
   {this.props.children}
</Fieldset>
```
