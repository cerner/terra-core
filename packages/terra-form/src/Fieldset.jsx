/* eslint-disable react/jsx-boolean-value, jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import 'terra-base/lib/baseStyles';

import './Field.scss';

const propTypes = {
  /**
   * Children the Field contains
   */
  children: PropTypes.node,
  /**
   * Error message for when the input is invalid
   */
  error: PropTypes.node,
  /**
   * Help element to display with the input
   */
  help: PropTypes.node,
  /**
   * Determines whether the fieldset is an inline fieldset
   */
  isInline: PropTypes.bool,
  /**
   * Legend for the input group
   */
  legend: PropTypes.string,
  /**
   * Determines whether the fieldset is required
   */
  required: PropTypes.bool,
};

const defaultProps = {
  children: null,
  error: null,
  help: null,
  isInline: false,
  legend: null,
  required: false,
};

const Fieldset = ({ children, error, help, isInline, legend, required, ...customProps }) => {
  const fieldsetClasses = classNames(
    'terra-Form-fieldset',
    { 'terra-Form-fieldset--inline': isInline },
    { 'terra-Form-fieldset--required': required },
    customProps.className,
  );

  return (
    <fieldset {...customProps} className={fieldsetClasses}>
      {legend && <legend className="terra-Form-legend">{legend}</legend>}
      {children}
      {help && <small className="terra-Form-helpText" tabIndex="-1">{help}</small>}
      {error && <small className="terra-Form-error" tabIndex="-1">{error}</small>}
    </fieldset>
  );
};

Fieldset.propTypes = propTypes;
Fieldset.defaultProps = defaultProps;

export default Fieldset;
