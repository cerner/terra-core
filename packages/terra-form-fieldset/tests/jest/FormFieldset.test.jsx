/* eslint-disable react/jsx-boolean-value */
import React from 'react';
import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';

import Fieldset from '../../src/Fieldset';

it('should render a default component', () => {
  const field = (<Fieldset />);
  const wrapper = shallow(field);
  expect(wrapper).toMatchSnapshot();
});

it('should render a Fieldset when all the possible props are passed into it', () => {
  const input = (
    <Fieldset
      legend="Text"
      className="fieldset-custom"
      legendAttrs={{ className: 'healtheintent-legend' }}
      help="This is a test input"
      required
    >
      <input type="radio" value="Test" />
    </Fieldset>
  );

  const wrapper = shallow(input);
  expect(wrapper).toMatchSnapshot();
});

it('correctly applies the theme context className', () => {
  const wrapper = mount(
    <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
      <Fieldset />
    </ThemeContextProvider>,
  );
  expect(wrapper).toMatchSnapshot();
});
