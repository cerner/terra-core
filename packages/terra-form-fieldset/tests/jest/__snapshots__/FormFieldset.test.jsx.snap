// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly applies the theme context className 1`] = `
<ThemeContextProvider
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <Fieldset
    isLegendHidden={false}
    legendAttrs={Object {}}
    required={false}
  >
    <fieldset
      className="fieldset orion-fusion-theme"
    >
      <div
        className="fieldset-children"
      />
    </fieldset>
  </Fieldset>
</ThemeContextProvider>
`;

exports[`should render a Fieldset when all the possible props are passed into it 1`] = `
<fieldset
  className="fieldset fieldset-required fieldset-custom"
>
  <legend
    className="legend healtheintent-legend"
  >
    Text
  </legend>
  <small
    className="help-text"
    tabIndex="-1"
  >
    This is a test input
  </small>
  <div
    className="fieldset-children"
  >
    <input
      type="radio"
      value="Test"
    />
  </div>
</fieldset>
`;

exports[`should render a default component 1`] = `
<fieldset
  className="fieldset"
>
  <div
    className="fieldset-children"
  />
</fieldset>
`;
